packages/fxa-settings/src/lib/metrics.ts

```diff
@@ -0,0 +1,377 @@
+ */
+export function init(
+  /** Flow data sent via query params from the content-server */
+  _flowEventData: FlowEventData
+) {
+  if (!initialized) {
+    if (
+      _flowEventData.deviceId &&
+      _flowEventData.flowBeginTime &&
+      _flowEventData.flowId
+    ) {
+      flowEventData = _flowEventData;
+      initialized = true;
+    } else {
+      let redirectPath = window.location.pathname;
+      if (window.location.search) {
+        redirectPath += window.location.search;
+      }
+
+      return window.location.replace(
+        `${window.location.origin}/get_flow?redirect_to=${encodeURIComponent(
+          redirectPath
+        )}`
+      );
+    }
+  }
+}
+
+/**
+ * Set the value of a configurable metrics event property
+ */
+export function setProperty(
+  /** Property to set */
+  key: keyof ConfigurableProperties,
+  /** Property's value */
+  value: ConfigurableProperties[typeof key]
+) {
+  // FIXME: Type 'string | number | boolean | UserPreferences | ExperimentGroup[] |
+  //   MarketingCampaign[] | string[]' is not assignable to type 'never'.
+  // @ts-ignore
+  configurableProperties[key] = value;
+}
+
+/**
+ * Set the value of multiple configurable metrics event properties
+ */
+export function setProperties(
+  // Properties + values to configure
+  properties: Hash<any>
+) {
+  configurableProperties = Object.assign(configurableProperties, properties);
+}
+
+/**
+ * Set the view name prefix for metrics that contain a viewName.
+ * This is used to differentiate between flows when the same
+ * URL can appear in more than one place in the flow, e.g., the
+ * /sms screen. The /sms screen can be displayed in either the
+ * signup or verification tab, and we want to be able to
+ * differentiate between the two.
+ *
+ * This prefix is prepended to the view name anywhere a view
+ * name is used.
+ */
+export function setViewNamePrefix(
+  /** Desired view name prefix */
+  prefix: string
+) {
+  viewNamePrefix = prefix;
+}
+
+/**
+ * Initialize a payload of metric event data to the metrics endpoint
+ */
+export function logEvents(
+  /** Events to log */
+  eventSlugs: string[] = [],
+  /* Additional properties to log with the events */
+  eventProperties: Hash<any> = {}
+) {
+  try {
+    const now = Date.now();
+    const eventOffset = now - flowEventData.flowBeginTime!;
+
+    postMetrics(
+      Object.assign(configurableProperties, {
+        duration: 0, // TODO where is this set?
+        events: eventSlugs.map((slug) => ({ type: slug, offset: eventOffset })),
+        flushTime: now,
+        referrer: window.document.referrer || NOT_REPORTED_VALUE,
+        screen: getScreenInfo(),
+        ...flowEventData,
+        ...eventProperties,
+      })
+    );
+  } catch (e) {
+    console.error('AppError', e);
+    sentryMetrics.captureException(e);
+  }
+}
+
+/**
+ * Log an event with the view name as a prefix
+ */
+export function logViewEvent(
+  /** Name of the view */
```

Comment:
please don't comment self-evident parameters

Reply:
I can see why it's redundant and I'll remove, but I generally lean toward more documentation than less; if anything the comment could be updated to be more descriptive.

Reply:
We should strive for consistent comment style across the project. Since most of our packages were separate and maintained by different people we've got a long way to go, but if you'd like to propose a target style, please do it in a form we can discuss as a team. I don't want to have a mix of everyone's individual styles. In the meantime try to roughly match the rest of the project. We've got more function header comments with param descriptions than inline in the declaration, for example.

Reply:
I totally agree - consistent comment style is important, and I can definitely update these code comments to live at function headers where appropriate. However I would still argue that a more descriptive argument comment is better than none at all. I do concede that my comment in its original form was redundant, though.